modules/swagger-parser/src/main/java/io/swagger/parser/SwaggerParser.java

```diff
@@ -47,19 +47,30 @@ public Swagger read(String location, List<AuthorizationValue> auths, boolean res
         return null;
     }
 
+    private static Swagger getSwagger(String swaggerAsString) {
```

Comment:
It seems to me there is one parse method too many - why not just put this code inside the parse method below?